src/flows/components/header/index.tsx

```diff
@@ -133,13 +108,31 @@ const FlowHeader: FC = () => {
       .catch(err => console.error('failed to delete share', err))
   }
 
-  const generateLink = () => {
+  const createAndGetReadOnlyToken = async (): Promise<string> => {
+    const authName = `${flow.name} ${flow.createdAt}`
+    const response = await createReadOnlyAllAuthorization(orgID, authName)
+
+    return response.token
+  }
+
+  const generateLink = async () => {
     setLinkLoading(RemoteDataState.Loading)
+    let token: string
+
+    try {
+      token = await createAndGetReadOnlyToken()
+    } catch (e) {
+      console.error('failed to create read-only token')
+      setLinkLoading(RemoteDataState.Error)
+
+      return
```

Comment:
i think my previous comment got removed by a force push, but shouldn't we be setting a `setLinkLoading` to Done at some point? Also, not sure why we're storing the `token` if it's never being used. Lastly, should we send an event when the read-only token fails to generate?

Reply:
I didn't do a force push. So, not sure what happened to your comment.

`setLinkLoading` is getting set to done [here](https://github.com/influxdata/ui/pull/3702/files#diff-b8e90c3282c7a31096b6bbfc1a34c6bb24040e637e7fcafbd1cecc5942b0210aR140) after this in `postNotebooksShare.then` and it's being set to error in `catch`..

Reply:
ah that comment is still there.. on the resolved comment.

Reply:
K, that checks out. What about sending an event when the notebook fails to generate a token rather than just swallowing the error in a console.error?